src/auth/googleauth.ts

```diff
@@ -44,7 +45,11 @@ import {AuthClient} from './authclient';
  * Defines all types of explicit clients that are determined via ADC JSON
  * config file.
  */
-export type JSONClient = JWT | UserRefreshClient | BaseExternalAccountClient;
+export type JSONClient =
+  | JWT
+  | UserRefreshClient
+  | BaseExternalAccountClient
+  | Impersonated;
```

Comment:
We will want to make sure that this doesn't cause compilation errors for existing libraries like `googleapis` and `nodejs-secret-manager`.

Depending, we may need to treat this as a major change.

Reply:
I linked against this, and it seemed fine.